app/src/full/java/io/homeassistant/companion/android/sensors/AndroidAutoSensorManager.kt

```diff
@@ -25,29 +46,105 @@ class AndroidAutoSensorManager : SensorManager, Observer<Int> {
+            io.homeassistant.companion.android.common.R.string.sensor_description_android_auto_car_name,
+            "mdi:car-info"
+        )
+        private val carStatus = SensorManager.BasicSensor(
+            "android_auto_car_status",
+            "sensor",
+            io.homeassistant.companion.android.common.R.string.basic_sensor_name_android_auto_car_status,
+            io.homeassistant.companion.android.common.R.string.sensor_description_android_auto_car_status,
+            "mdi:ev-station",
+            deviceClass = "plug"
+        )
+        private val odometerValue = SensorManager.BasicSensor(
+            "android_auto_odometer",
+            "sensor",
+            io.homeassistant.companion.android.common.R.string.basic_sensor_name_android_auto_odometer,
+            io.homeassistant.companion.android.common.R.string.sensor_description_android_auto_odometer,
+            "mdi:map-marker-distance",
+            unitOfMeasurement = "m",
+            stateClass = SensorManager.STATE_CLASS_TOTAL_INCREASING,
+            deviceClass = "distance"
+        )
+        private var alreadyConnected = false
+
+        private val sensorsList = listOf(androidAutoConnected, batteryLevel, carName, carStatus, fuelLevel, odometerValue)
+        private enum class Listener {
+            ENERGY, MODEL, MILEAGE, STATUS,
+        }
+        private val listenerSensors = mapOf(
+            Listener.ENERGY to listOf(batteryLevel, fuelLevel),
+            Listener.MODEL to listOf(carName),
+            Listener.STATUS to listOf(carStatus),
+            Listener.MILEAGE to listOf(odometerValue)
+        )
+        private val listenerLastRegistered = mutableMapOf(
+            Listener.ENERGY to -1L,
+            Listener.MODEL to -1L,
+            Listener.STATUS to -1L,
+            Listener.MILEAGE to -1L
+        )
     }
 
     override val name: Int
         get() = commonR.string.sensor_name_android_auto
 
     override suspend fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
         return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            listOf(androidAutoConnected)
+            sensorsList
         } else {
             emptyList()
         }
     }
 
     override fun requiredPermissions(sensorId: String): Array<String> {
-        return emptyArray()
+        return when {
+            (sensorId == fuelLevel.id || sensorId == batteryLevel.id) -> {
+                arrayOf("com.google.android.gms.permission.CAR_FUEL")
+            }
+            sensorId == odometerValue.id -> {
+                arrayOf("com.google.android.gms.permission.CAR_MILEAGE")
+            }
+            else -> emptyArray()
+        }
     }
 
     private lateinit var context: Context
     private var carConnection: CarConnection? = null
 
+    private fun allDisabled(): Boolean {
+        return !sensorsList.map { x -> isEnabled(context, x) }.contains(true)
+    }
```

Comment:
Using a few Kotlin features

```suggestion
    private fun allDisabled(): Boolean =
        sensorsList.none { isEnabled(context, it) }
```